lib/street_address.rb

```diff
@@ -852,7 +852,7 @@ def state_name
       end
 
       def intersection?
-        !street2.nil?
+        !!street && !!street2
```

Comment:
Tricky rails stuff, but if I just did `street && street2`, the output value of `intersection?` was actually a string value (albeit, truthy), and that broke the assertions of some tests. For that reason, went with `!!`.

Reply:
totally up to you, but you could update those tests to `assert addr.intersection?` and `refute addr.intersection?` where it is and is not an intersection, respectively.

Reply:
Out of principle, I feel like a `?` method should return a boolean and that was the expectation before, so I'd personally go with a statement that holds to that rather than change the tests. 🤷‍♂️